### ISSUE TITLE
jsonnet std.parseYaml wraps result in an array when string values contain ---

### ISSUE THREAD

**Issue by SamuraiPrinciple** [argoproj/argo-cd#17676]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When using Jsonnet to generate manifests, std.parseYaml function returns incorrect result when string values contain `---`

I've also filed an issue here:

https://github.com/google/jsonnet/issues/1147

Jsonnet go implementation (https://github.com/google/go-jsonnet) seems unaffected when I tested it locally (which is strange, because I believe that's the version argocd is using).

**To Reproduce**

Create an app with `test.jsonnet` file with the following content:

```
{
  apiVersion: 'v1',
  kind: 'ConfigMap',
  metadata: { name: 'repro' },
  data: {
    'key': std.parseYaml('a: bad---'),
  },
}
```

**Expected behavior**

the `key` value should be ```{"a": "bad---"}``` and not ```[{"a": "bad---"}]```

**Screenshots**

<img width="2049" alt="Screenshot 2024-03-31 at 15 43 54" src="https://github.com/argoproj/argo-cd/assets/621249/fc994bfc-e5fe-43e7-b575-de1f14e6612b">

**Version**

```shell
argocd: v2.10.5+335875d
  BuildDate: 2024-03-28T15:02:45Z
  GitCommit: 335875d13e018bed6e03873f4742582582964745
  GitTreeState: clean
  GoVersion: go1.21.3
  Compiler: gc
  Platform: linux/amd64
FATA[0000] Argo CD server address unspecified           
```

**Logs**

Not relevant

**Comment by SamuraiPrinciple** [comment]
Apologies, after receiving some feedback on issue https://github.com/google/jsonnet/issues/1147 I've realised that I've got it wrong way around - it's the go-jsonnet version that affected (which explains why argocd is affected too).

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and tell us if the issue is still present, please?

**Comment by SamuraiPrinciple** [comment]
Yes, this is still the case with the latest version (2.13.1) - this snippet:

{
  apiVersion: 'v1',
  kind: 'ConfigMap',
  metadata: { name: 'repro', namespace: 'ingress-nginx-test' },
  data: {
    key: std.manifestYamlDoc(std.parseYaml('a: bad---')),
  },
}


would emit:

apiVersion: v1
data:
  key: '- "a": "bad---"'
kind: ConfigMap
metadata:
  labels:
    argocd.argoproj.io/instance: cluster-bootstrap
  name: repro
  namespace: ingress-nginx-test

instead of:

apiVersion: v1
data:
  key: '"a": "bad---"'
kind: ConfigMap
metadata:
  labels:
    argocd.argoproj.io/instance: cluster-bootstrap
  name: repro
  namespace: ingress-nginx-test
